test/spec/linkify-string-test.js

```diff
@@ -71,9 +72,44 @@ describe('linkify-string', function () {
 		});
 	});
 
-	it('Works with overriden options', function () {
+	it('Works with overriden options (general)', function () {
 		tests.map(function (test) {
 			expect(linkifyStr(test[0], options)).to.be.eql(test[2]);
 		});
 	});
+
+	// Test specific options
+	it('Works with overriden options (validate)', function () {
+		var options_validate = {
+			validate: function (hasProtocol, text, type) {
+				return type === 'email' || (hasProtocol || text.slice(0,3) === 'www');
```

Comment:
I would change the first check here to `type !== 'url'`, since there could conceivably be tokens other than `email`